lib/validate-config.js

```diff
@@ -56,6 +56,51 @@ function validateConfig(config, configPath) {
 
   const configWithDefaults = Object.assign({}, defaults, config);
 
+  const validConfigProperties = [
+    'siteBasePath',
+    'siteOrigin',
+    'wrapperPath',
+    'notFoundPath',
+    'externalStylesheets',
+    'autoprefixerBrowsers',
+    'pagesDirectory',
+    'staticDirectory',
+    'outputDirectory',
+    'temporaryDirectory',
+    'data',
+    'dataSelectors',
+    'vendorModules',
+    'webpackLoaders',
+    'webpackPlugins',
+    'babelPlugins',
+    'babelExclude',
+    'postcssPlugins',
+    'fileLoaderExtensions',
+    'jsxtremeMarkdownOptions',
+    'inlineJs',
+    'production',
+    'port',
+    'verbose'
+  ];
```

Comment:
Let's pull this list out of the function, since it doesn't have to be created when the function executes — but is a kind of constant. Also, let's add a comment noting that whenever you add a new configuration property you'll need to add to this list.